zone/raids.cpp

```diff
@@ -2203,3 +2257,734 @@ void Raid::SetNewRaidLeader(uint32 i)
 		}
 	}
 }
+
+void Raid::SaveRaidNote(const char* who, const char* note) {
+
+	if (strlen(who) == 0 || strlen(note) == 0) {
+		return;
+	}
+
+	std::string query = StringFormat("UPDATE raid_members SET note = '%s' WHERE raidid = %lu AND name = '%s'",
+		Strings::Escape(note).c_str(),
+		GetID(),
+		Strings::Escape(who).c_str()
+	);
+	auto results = database.QueryDatabase(query);
+
+}
+
+std::vector<RaidMember> Raid::GetMembersWithNotes()
+{
+	std::vector<RaidMember> raid_members;
+	for (const auto& m : members) {
+		if (strlen(m.note) != 0) {
+			raid_members.emplace_back(m);
+		}
+	}
+	return raid_members;
+}
+
+void Raid::SendRaidNotes()
+{
+	LearnMembers();
+	VerifyRaid();
+
+	for (const auto& c : GetMembersWithNotes()) {
+		auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(RaidGeneral_Struct));
+		auto note = (RaidGeneral_Struct*)outapp->pBuffer;
+		note->action = raidSetNote;
+		strn0cpy(note->leader_name, c.member_name, 64);
+		strn0cpy(note->player_name, GetLeaderName().c_str(), 64);
+		strn0cpy(note->note, c.note, 64);
+		QueuePacket(outapp);
+		safe_delete(outapp);
+	}
+}
+void Raid::SendRaidNotesToWorld()
+{
+	auto pack = new ServerPacket(ServerOP_RaidNote, sizeof(ServerRaidNote_Struct));
+	auto snote = (ServerRaidNote_Struct*)pack->pBuffer;
+	snote->rid = GetID();
+	worldserver.SendPacket(pack);
+	safe_delete(pack);
+}
+
+void Raid::DelegateAbilityAssist(Mob* delegator, const char* delegatee)
+{
+	auto raid_delegatee = entity_list.GetRaidByName(delegatee);
+	if (!raid_delegatee) {
+		delegator->CastToClient()->MessageString(Chat::Cyan, 5082, delegatee);
+		return;
+	}
+	uint32 raid_delegatee_id = raid_delegatee->GetID();
+	uint32 raid_delegator_id = GetID();
+	if (raid_delegatee_id != raid_delegator_id) {
+		delegator->CastToClient()->MessageString(Chat::Cyan, NOT_IN_YOUR_RAID, delegatee);
+		return;
+	}
+
+	auto rm = &members[GetPlayerIndex(delegatee)];
+	if (!rm) { 
+		return; 
+	}
+	auto c = rm->member;
+	if (!c) { 
+		return; 
+	}
+
+	auto slot = FindNextRaidDelegateSlot(FindNextAssisterSlot);
+	auto ma = rm->main_assister;
+	if (slot == -1 && !ma) {
+		delegator->CastToClient()->MessageString(Chat::Cyan, MAX_MAIN_RAID_ASSISTERS);
+		return;
+	}
+
+	auto outapp = new EQApplicationPacket(OP_RaidDelegateAbility, sizeof(DelegateAbility_Struct));
+	DelegateAbility_Struct* das = (DelegateAbility_Struct*)outapp->pBuffer;
+	if (ma) {
+		das->Action = ClearDelegate;
+		memset(main_assister_pcs[ma - 1], 0, 64);
+		rm->main_assister = 0;
+		std::string query = StringFormat("UPDATE raid_members SET is_assister = 0 WHERE raidid = %i AND name= '%s';",
```

Comment:
Stuff like this could be made in to a repository method in `common/repositories/raid_members_repository.h`.

Example being this.
```cpp
static int UpdateRaidAssister(
	Database& db,
	int32_t raid_id,
	const std::string& character_name,
	int8 assister
) {
	auto results = db.QueryDatabase(
		fmt::format(
			"UPDATE {} SET is_assister = {} WHERE raidid = {} AND `name` = '{}'",
			TableName(),
			assister,
			raid_id,
			Strings::Escape(character_name)
		)
	);

	return results.Success() ? results.RowsAffected() : 0;
}
```

Then all you have to do is this.
```cpp
int updated = RaidMembersRepository::UpdateRaidAssister(database, GetID(), delegatee, 0);
if (!updated) {
    LogError("Unable to clear raid main assister for player [{}]", delegatee);
    return;
}
```

Reply:
Thanks for this suggestion!  I was looking at this previously though decided to stick to the existing database pattern.  I have now updated all my calls to use the repositories.  I will use them going forward for future work.